accord-core/src/main/java/accord/utils/SimpleBitSet.java

```diff
@@ -0,0 +1,348 @@
+            Invariants.checkArgument(to >= from, "to < from (%s < %s)", to, from);
+            return;
+        }
+
+        int fromIndex = from >>> 6;
+        int toIndex = (to + 63) >>> 6;
+        if (fromIndex + 1 == toIndex)
+        {
+            long addBits = (-1L >>> (64 - (to & 63))) & (-1L << (from & 63));
+            orBitsAtIndex(fromIndex,  addBits);
+        }
+        else if (count == 0)
+        {
+            bits[toIndex - 1] = -1L >>> (64 - (to & 63));
+            for (int i = fromIndex + 1, maxi = toIndex - 1; i < maxi ; ++i)
+                bits[i] = -1L;
+            bits[fromIndex] = -1L << (from & 63);
+            count = to - from;
+        }
+        else
+        {
+            orBitsAtIndex(fromIndex, -1L << (from & 63));
+            for (int i = fromIndex + 1, maxi = toIndex - 1; i < maxi ; ++i)
+            {
+                count += 64 - Long.bitCount(bits[i]);
+                bits[i] = -1L;
+            }
+            orBitsAtIndex(toIndex - 1, -1L >>> (64 - (to & 63)));
+        }
+    }
+
+    private void orBitsAtIndex(int index, long setBits)
+    {
+        long prevBits = bits[index];
+        bits[index] = setBits | prevBits;
+        count += Long.bitCount(setBits) - Long.bitCount(prevBits);
+    }
+
+    public boolean unset(int i)
+    {
+        int index = indexOf(i);
+        long bit = bit(i);
+        if (0 == (bits[index] & bit))
+            return false;
+        bits[index] &= ~bit;
+        --count;
+        return true;
+    }
+
+    public boolean get(int i)
+    {
+        int index = indexOf(i);
+        long bit = bit(i);
+        return 0 != (bits[index] & bit);
+    }
+
+    public int size()
+    {
+        return bits.length * 64;
+    }
+
+    public int setBitCount()
+    {
+        return count;
+    }
+
+    public boolean isEmpty()
+    {
+        return count == 0;
+    }
+
+    public int prevSetBit(int i)
+    {
+        if (count == 0)
+            return -1;
+
+        int index = indexOf(i);
+        long bits = this.bits[index] & bitsEqualOrLesser(i);
+        while (true)
+        {
+            if (bits != 0)
+                return index * 64 + numberOfTrailingZeros(highestOneBit(bits));
+
+            if (--index < 0)
+                return -1;
+
+            bits = this.bits[index];
+        }
+    }
+
+    public int prevSetBitNotBefore(int i, int inclBound, int ifNotFound)
+    {
+        if (count == 0)
+            return ifNotFound;
+
+        int index = indexOf(i);
+        int inclIndexBound = lowerLimitOf(inclBound);
+        long bits = this.bits[index] & bitsEqualOrLesser(i);
+        while (true)
+        {
+            if (bits != 0)
+            {
+                int result = index * 64 + numberOfTrailingZeros(highestOneBit(bits));
+                return result > inclBound ? result : ifNotFound;
+            }
+
+            if (--index < inclIndexBound)
+                return -1;
+
+            bits = this.bits[index];
+        }
```

Comment:
```suggestion
        int result = prevSetBit(i);
        return result > inclBound ? result : ifNotFound;
```

Reply:
if you really want to exit early in

```
if (--index < 0)
                return -1;
```

can change `prevSetBit` to delegate to `prevSetBitNotBefore` instead...